### ISSUE TITLE
[Bug] MVTLayer updateTrigger no longer working inside Docker container

### ISSUE THREAD

**Issue by mplatt212** [visgl/deck.gl#8029]
### Description

I have a MVTLayer inside of a React project residing in a Docker container.  I have a ternary statement which decides the onFillColor and updateTriggers with the variable to update on.  Basically if I hover on a country the selected country changes thus triggering the fill color to change.  Everything was working great until one day it just didn't anymore (I can't pinpoint the exact change where things went downhill).  The updateTrigger no longer seems to work properly and the fill color does not change anymore.  It still works for everyone on my team and the strange thing is that it works for me outside of Docker and also it works in Docker if I use a GeoJsonLayer set up the same way.  Also, the onHover and autoHighlight functions still work for the MVTLayer.  I've tried changing branches, reseting in Git but nothing seems to work.  Any insight here as to what might be the cause of this?  I've been stuck for days.

### Flavors

- [ ] Script tag
- [X] React
- [ ] Python/Jupyter notebook
- [ ] MapboxOverlay
- [ ] GoogleMapsOverlay
- [ ] CartoLayer
- [ ] ArcGIS

### Expected Behavior

I expect to see the fill color change for whichever country I hover over.

### Steps to Reproduce

-Use Docker to containerize React app with Deck.gl.
-Add MVTLayer to DeckGL
-Use updateTriggers and getFillColor to change color of asset on hover.

```/* DEPENDENCIES */
import React, { useEffect, useState } from 'react';
import { DeckGL } from '@deck.gl/react';
import MapGL, { ScaleControl } from 'react-map-gl';
import maplibregl from 'maplibre-gl';
import { observer } from 'mobx-react-lite';
import { useStore } from '../../store/Provider';
import { MVTLayer } from 'deck.gl';
interface IProps {
  //   expandMap: boolean;
}

const Map = observer((props: IProps) => {
  const store = useStore();
  const [tileUrl, setTileUrl] = useState<string>('');

  const mapStyle = {
    version: 8,
    sources: {
      'raster-tiles': {
        type: 'raster',
        // tiles: ,
        tiles: [
          'https://a.tile.openstreetmap.org/{z}/{x}/{y}.png',
          'https://b.tile.openstreetmap.org/{z}/{x}/{y}.png',
          'https://c.tile.openstreetmap.org/{z}/{x}/{y}.png',
        ],
        tileSize: 256,
      },
    },
    layers: [
      {
        id: 'osm-tiles',
        type: 'raster',
        source: 'raster-tiles',
        minzoom: 0,
        maxzoom: 20,
      },
    ],
  };

  const ICON_MAPPING = {
    marker: { x: 0, y: 0, width: 128, height: 128, mask: true },
  };

  const vectorTiles = new MVTLayer({
      id: 'MVTLayer_1',
      data: tileUrl,
      pickable: true,
      // @ts-ignore
      pointType: "icon",
      iconAtlas:
        "https://raw.githubusercontent.com/visgl/deck.gl-data/master/website/icon-atlas.png",
      iconMapping: ICON_MAPPING,
      getIcon: (x: any) => "marker",
      lineWidthScale: 20,
      lineWidthMinPixels: 2,
      getFillColor: (d: any) =>
        store.mapStore.selectedCountry &&
        d.properties.iso_a3 === store.mapStore.selectedCountry?.iso_a3
          ? [0, 100, 0, 100]
          : [0, 0, 50, 100],
      getLineColor: [100, 0, 0, 255],
      getLineWidth: 10,
      getElevation: 30,
      onClick: (e) => {
        console.log(e);
        store.mapStore.setShowModal(true);
        console.log(
          store.mapStore.showModal,
          store.mapStore.selectedCountry?.iso_a3
        );
      },
      onHover: (el: any) => {
        if (el.object?.properties) {
          store.mapStore.setSelectedCountry(el.object.properties);
          store.mapStore.setCursor('crosshair');
          store.mapStore.setHoverInfoCoords(el.pixel);
          store.mapStore.setHoverInfo(
            `Country:  ${el.object?.properties.admin}  iso_a3:  ${el.object?.properties.iso_a3}`
          );
        } else {
          store.mapStore.setCursor('pointer');
          store.mapStore.setHoverInfo('');
        }
      },
      updateTriggers: {
        getFillColor: store.mapStore.selectedCountry,
      },
    });
    
    const getVectorTiles = async () => {
      try {
        await fetch(`http://localhost:7026/backend/api/proxy/tileserv/psadmin.countries`, {})
          .then((r) => r.json())
          .then((data) => setTileUrl(data.tileurl));
      } catch (ex) {
        console.log(ex);
      }
    };

    useEffect(() => {
      getVectorTiles();
    }, [])

  return (
    <DeckGL
      layers={[vectorTiles]}
      controller={true}
      initialViewState={store.mapStore.viewState}
      // onViewStateChange={(el: any) => console.log(el)}
      getCursor={(x) => store.mapStore.cursor}
    >
      <MapGL
        height="100vh"
        width="100vw"
        mapLib={maplibregl}
        //@ts-ignore
        mapStyle={mapStyle}
      >
        <ScaleControl position="bottom-left" unit="imperial" />
      </MapGL>
      {store.mapStore.hoverInfo && (
        <div
          style={{
            position: 'absolute',
            zIndex: 1,
            pointerEvents: 'none',
            top: store.mapStore.hoverInfoCoords[1],
            left: store.mapStore.hoverInfoCoords[0],
            color: '#fffaf1',
          }}
        >
          {store.mapStore.hoverInfo}
        </div>
      )}
    </DeckGL>
  );
});

export default Map;```

### Environment

- Framework version: React ^18.2
- Browser: Chrome
- OS: Windows 10
- DeckGL ^8.9.6


### Logs

Not sure if these are relevant:
![image](https://github.com/visgl/deck.gl/assets/70013024/9ded8588-1c61-4415-be7c-7f1d62a98dff)

See this inside Docker logs:
![image](https://github.com/visgl/deck.gl/assets/70013024/8b0c3d6a-2b99-46eb-8dcb-b77f12a0a5b7)

**Comment by Pessimistress** [comment]
A few suggestions:

- [Format your code](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#quoting-code) properly
- Find out the exact deck.gl version installed by `npm ls` or `yarn list`. In fact, that would be the first thing I check between your local environment and the docker image.
- If you cannot reproduce the issue outside of your container, then I'm afraid we cannot reproduce it either. You should move to Discussions instead of reporting a bug.

**Comment by mplatt212** [comment]
@Pessimistress Apologies, code is now formatted.  The app locally and inside the container are both on deck.gl@8.9.22.  Is there a way to move this issue to discussions or shall I just close and copy/paste it?

**Comment by Pessimistress** [comment]
Just close it. Feel free to reopen if/when you do have a repro.

**Comment by mplatt212** [comment]
I am reopening this issue as it turns out this can be reproduced outside the Docker container.  For some reason 'npm install' inside docker with deck.gl@^8.9.6 in the package.json was installing 8.9.22 but locally it was installing 8.9.7.  Whenever I explicitly install version 8.9.20 or above it seems updateTriggers with getFillColor do not function as expected.  I didn't mention earlier but this is within the Gatsby framework.

**Comment by Pessimistress** [comment]
Fixed in v8.9.23.
